invenio_rdm_records/ext.py

```diff
@@ -256,3 +277,16 @@ def fix_datacite_configs(self, app):
         for config_item in datacite_config_items:
             if config_item in app.config:
                 app.config[config_item] = str(app.config[config_item])
+
+    def validate_config(self, app):
+        """Validate the configuration."""
+        token_length = app.config["RDM_RECORDS_GUEST_ACCESS_TOKEN_LENGTH"]
+        if token_length is not None and token_length <= 0:
+            raise Exception("invalid RDM_RECORDS_GUEST_ACCESS_TOKEN_LENGTH")
+
+        link_permission_levels = app.config["RDM_RECORDS_SECRET_LINK_PERMISSION_LEVELS"]
+        grant_permission_levels = app.config["RDM_RECORDS_GRANT_PERMISSION_LEVELS"]
+        if not set(link_permission_levels).intersection(set(grant_permission_levels)):
+            raise Exception(
+                "no intersection between RDM_RECORDS_SECRET_LINK_PERMISSION_LEVELS and RDM_RECORDS_GRANT_PERMISSION_LEVELS"
```

Comment:
Because the access requests generate **grants** for the user-based workflow and **secret links** for the guest-based workflow.
This is only really "necessary" if the access request settings apply to both workflows.
If we handle them separately (separate setting for the default permission level given by an accepted `guest-access-request` and `user-access-request`), there's no need for an overlap anymore.

Reply:
That being said, I've removed the configurability of the available permission levels from the PR for now, to reduce the scope.